python-avd/pyavd/_eos_designs/structured_config/network_services/vxlan_interface.py

```diff
@@ -24,152 +26,98 @@ class VxlanInterfaceMixin(Protocol):
     """
 
     @cached_property
-    def vxlan_interface(self: AvdStructuredConfigNetworkServicesProtocol) -> dict | None:
+    def _multi_vtep(self: AvdStructuredConfigNetworkServicesProtocol) -> bool:
+        return self.shared_utils.mlag is True and self.shared_utils.evpn_multicast is True
+
+    @structured_config_contributor
+    def vxlan_interface(self: AvdStructuredConfigNetworkServicesProtocol) -> None:
         """
-        Returns structured config for vxlan_interface.
+        Update structured config for vxlan_interface.
 
         Only used for VTEPs and for WAN
 
         This function also detects duplicate VNIs and raise an error in case of duplicates between
         all Network Services deployed on this device.
         """
         if not (self.shared_utils.overlay_vtep or self.shared_utils.is_wan_router):
-            return None
+            return
 
-        vxlan = {
-            "udp_port": 4789,
-        }
+        vxlan = EosCliConfigGen.VxlanInterface.Vxlan1.Vxlan(udp_port=4789)
```

Comment:
Any reason you want to keep a copy of all the data here instead of just inserting directly into self.structured_config everywhere? (By inserting directly we will not overwrite config contributed by other modules some time in the future.